src/Integration.Vsix/CFamily/ProcessRunner.cs

```diff
@@ -116,66 +119,87 @@ public bool Execute(ProcessRunnerArguments runnerArgs)
                     else
                     {
                         LogMessage(CFamilyStrings.MSG_ExecutionCancelled);
-                        return false;
+                        return;
                     }
                 }
 
-                var result = WaitForProcessToFinish(process, runnerArgs);
-                return result;
+                using (var timeoutTask = Task.Factory.StartNew(() => WaitForProcessToFinish(process, runnerArgs)))
+                {
+                    process.BeginErrorReadLine();
+
+                    // Warning: do not log the raw command line args as they
+                    // may contain sensitive data
+                    LogDebug(CFamilyStrings.MSG_ExecutingFile,
+                        runnerArgs.ExeName,
+                        runnerArgs.AsLogText(),
+                        runnerArgs.WorkingDirectory,
+                        runnerArgs.TimeoutInMilliseconds,
+                        process.Id);
+
+                    try
+                    {
+                        runnerArgs.HandleInputStream(process.StandardInput);
+                        runnerArgs.HandleOutputStream(process.StandardOutput);
+                    }
+                    catch (Exception) when (isRunningProcessCancelled)
+                    {
+                        // If a process is cancelled mid-stream, an exception will be thrown.
+                    }
+                    finally
+                    {
+                        timeoutTask.Wait();
```

Comment:
Why do we need to wait for the timeout task here? If I understand correctly, we'll already have read to the end of the output stream so we won't have any other processing to do.